akka-bench-jmh/src/main/scala/akka/remote/compress/HeavyHittersBenchmark.scala

```diff
@@ -3,55 +3,64 @@
  */
 package akka.remote.compress
 
-import java.util.{PriorityQueue, Random}
-import java.util.concurrent.ThreadLocalRandom
+import java.util.Random
 
 import akka.remote.artery.compress.TopHeavyHitters
 import org.openjdk.jmh.annotations._
+import org.openjdk.jmh.infra.Blackhole
```

Comment:
Right, one should either return the result of the benchmark (then JMH does the blackhole), or if it's not possible for some reason blackhole yourself.
It mostly is about a volatile write but not only, it does some more magic to disalow the jit to optimise stuff away there